some proposed mechanism for how to track file indirection via IANA registry (#49)
mjethanandani/veloce: draft-ietf-opsawg-veloce-yang.md

In place of including a YANG module, SID file or YANG Tree diagram into an Internet-Draft or RFC by value (via CDATA in the XML, for instance), an indirection is added that indirects through the IANA maintained table.

(XXX: YANG Tree diagrams are quite instructive, and compact give a very good overview of the structure of a module.  There are strong reasons to include them by value in a document.  On the other Hand: they depend directly on the module and would change with the module.  The recommendation here is to include the current version by value,
marked non-normative, and include the reference to the tree file via the registry as an inline normative reference)

The Lifecycle of the references is detailed below.
In order to provide the best introduction to the life cycle, the process will be started from the state of a published RFC with a published MODULE, and then follow the update process through a life cycle of updates.  Following that, the process for a new module is explained.

### Revision of a Module

An individual (or an individual given an action by a WG) goes to the repository given by the "latest" entry for the module revision.

> The module and related files are forked to a new git repository, on a new branch.

Author: ```suggestion
A new branch created for the revisions.  (This may be in a new related repository, forked, depending upon permissions)
```

The files are edited as desired and then an annotated (possibly signed) git tag is created, creating a new stable reference.

### Registration the new proposed module

IANA will be contacted, and a template (TBD below) is filled out to add a new entry.
The filetag and filetype will be the same as in the module that is being updated.

For versions of the form YYYY-MM-DD, then a new entry of the form YYYY-MM-DD.xx
will be proposed, where xx is an integer.  For Semantic Version X.Y.Z, then a new
X.Y.VV will be proposed with identical X and Y, and VV >= Z+1.

> Note that many revisions might require a higher X or Y when published, but that decision is deferred.

Author: Note that MCR is not really clued in to how YANG semver is gonna work.

(XXX: the form of the revision, and whether Semantic Version strings should be used at all here is open to debate.  Perhaps a new column would be better)

The entrytype is set to PROPOSED.
The URL is the new stable reference from above.
The controllingcontact is set by IANA to the proposer.
The sha256hash SHOULD be included by the proposer, but IANA is expected to retrieve the document and recalculate the hash.

### Official Revision of the module

When a YANG module is related to an RFC, then updates to it are controlled by a WG
Consensus Call.
The rational for the changes to the YANG module are to be described in a regular  internet-draft, and the WG SHOULD adopt that internet-draft like any other I-D.

> (The intended status of the internet-draft MAY be set to Experimental, but ideally, a new status will be added: "For IESG Approval Consideration")

Author: This is intentionally identical to what happens when the IESG is asked to redesignate an RFC from Proposed Standard to Internet Standard.    See for instance, https://datatracker.ietf.org/doc/draft-palet-v6ops-siit-dc-st/

> The I-D SHOULD reference the YANG module, SID file, and tree diagram using the proposed syntax below.   A normative reference

Author: ```suggestion
The I-D SHOULD reference the YANG module, SID file, and tree diagram using the proposed syntax below, as a normative reference.
```

> The WG process continues as for any I-D: reviews, WGLC, AD evaluation.
> Reviewers need to understand that they are reviewing the YANG modules (and SID files),
> not the contents of the I-D itself.

Author: ```suggestion
The WG process continues as for any I-D: reviews, WGLC, AD evaluation.
Reviewers need to understand that they are reviewing the YANG modules (and SID files),
not the contents of the I-D itself.  
There should be a standard template for the Introduction and Abstract as reviewers do not always clue in.
The I-D might omit Security Considerations, IANA Considerations.
```

# IANA Considerations

## File Indirection Registry {#filereferences}

IANA is asked to create a new registry, probably within the YANG Module Tags grouping:
https://www.iana.org/assignments/yang-module-tags/yang-module-tags.xhtml

> This new Registry is to updated according to details of each entry, explained below.

Author: Instead, we could add columns and changes rules on https://www.iana.org/assignments/yang-parameters#yang-parameters-1
